community/detectors/spring_framework_cve_2022_22965/src/main/java/com/google/tsunami/plugins/detectors/spring/SpringCve202222965Detector.java

```diff
@@ -0,0 +1,220 @@
+public final class SpringCve202222965Detector implements VulnDetector {
+
+  private static final GoogleLogger logger = GoogleLogger.forEnclosingClass();
+  private static final String VULNERABILITY_PAYLOAD_STRING_1 =
+      "class.module.classLoader.DefaultAssertionStatus=1";
+  private static final String VULNERABILITY_PAYLOAD_STRING_2 =
+      "class.module.classLoader.DefaultAssertionStatus=2";
+
+  private final Clock utcClock;
+  private final HttpClient httpClient;
+  private final SimpleCrawler simpleCrawler;
+
+  @Inject
+  SpringCve202222965Detector(@UtcClock Clock utcClock, HttpClient httpClient, Crawler crawler,
+      SimpleCrawler simpleCrawler) {
+    this.utcClock = checkNotNull(utcClock);
+    this.httpClient = checkNotNull(httpClient);
+    this.simpleCrawler = checkNotNull(simpleCrawler);
+  }
+
+  @Override
+  public DetectionReportList detect(
+      TargetInfo targetInfo, ImmutableList<NetworkService> matchedServices) {
+    return DetectionReportList.newBuilder()
+        .addAllDetectionReports(
+            matchedServices.stream()
+                .filter(NetworkServiceUtils::isWebService)
+                .filter(this::isServiceVulnerable)
+                .map(networkService -> buildDetectionReport(targetInfo, networkService))
+                .collect(toImmutableList()))
+        .build();
+  }
+
+  private static String buildTargetUrl(NetworkService networkService, String nextUri) {
+    StringBuilder targetUrlBuilder = new StringBuilder();
+    if (NetworkServiceUtils.isWebService(networkService)) {
+      targetUrlBuilder.append(NetworkServiceUtils.buildWebApplicationRootUrl(networkService));
+    } else {
+      // Assume the service uses HTTP protocol when the scanner cannot identify the actual service.
+      targetUrlBuilder
+          .append("http://")
+          .append(toUriAuthority(networkService.getNetworkEndpoint()))
+          .append("/");
+    }
+    targetUrlBuilder.append(nextUri);
+    return targetUrlBuilder.toString();
+  }
+
+  private static final ImmutableSet<String> PATHS = ImmutableSet.of(
+//      "/login.html",
+//      "/login",
+//      "/logout",
+//      "/success",
+//      "/fail",
+//      "/error",
+//      "/index",
+//      "/",
+      "/index.html"
+      );
+
+  private boolean isServiceVulnerable(NetworkService networkService) {
+    Set<String> crawlTargets = PATHS.stream()
+        .map(path -> buildTargetUrl(networkService, path)).collect(Collectors.toSet());
+    CrawlConfig crawlConfig = CrawlConfig.newBuilder()
+        .setMaxDepth(1)
+        .addAllSeedingUrls(crawlTargets).build();
+    ListenableFuture<ImmutableSet<CrawlResult>> crawlResultsFuture =
+        simpleCrawler.crawlAsync(crawlConfig);
+    ImmutableSet<CrawlResult> crawlResults = null;
+    try {
+      crawlResults = crawlResultsFuture.get();
+    } catch (Exception e) {
+      logger.atWarning().withCause(e).log("Unable to crawl.");
+      return false;
+    }
```

Comment:
You don't need to perform your own crawling in the detectors. Tsunami's fingerprinter will populate all the crawling results into the [`NetworkService.service_context.web_service_context.crawl_results` field](https://github.com/google/tsunami-security-scanner/blob/master/proto/network_service.proto#L78).